aptos-move/e2e-tests/src/executor.rs

```diff
@@ -988,33 +992,32 @@ impl FakeExecutor {
         self.exec_module(&Self::module(module_name), function_name, type_params, args)
     }
 
-    pub fn try_exec_entry_with_features(
+    pub fn try_exec_entry_with_resolver(
         &mut self,
         senders: Vec<AccountAddress>,
         entry_fn: &EntryFunction,
-        features: &Features,
+        resolver: &impl AptosMoveResolver,
     ) -> Result<(WriteSet, Vec<ContractEvent>), VMStatus> {
-        let resolver = self.data_store.as_move_resolver();
-
         let timed_features = TimedFeaturesBuilder::enable_all()
             .with_override_profile(TimedFeatureOverride::Testing)
             .build();
 
+        let mut features = Features::fetch_config(resolver).unwrap_or_default();
+        features.enable(FeatureFlag::VM_BINARY_FORMAT_V6);
+        let struct_constructors = features.is_enabled(FeatureFlag::STRUCT_CONSTRUCTORS);
         let vm = MoveVmExt::new(
             NativeGasParameters::zeros(),
             MiscGasParameters::zeros(),
             LATEST_GAS_FEATURE_VERSION,
             self.chain_id,
-            features.clone(),
+            features,
             timed_features,
-            &resolver,
+            resolver,
         )
         .unwrap();
-        let mut session = vm.new_session(&resolver, SessionId::void());
-
+        let mut session = vm.new_session_with_flush_flag(resolver, SessionId::void(), true);
```

Comment:
Hmm the logic indeed looks pretty weird to me. Ideally I would avoid making changes in the MoveVM but would like to understand the problem here a bit better. 

Looking at the code, the MoveVM will be spawned fresh in line 1008. The only thing that was cached previously is the `WarmVMCache` which would probably cache the previous version of Aptos Framework. Is that the code you were having issue with? i.e: the older version of aptos framework is being loaded and cause execution problem?

Reply:
Yeah... Currently, the tool executes v1 and then v2 for each txn. After execution of v1 code, we need to reload the aptos framework compiled by v2.

Reply:
My take is we should avoid making changes here in the Move VM. Instead, we should look into how to invalidate this WarmVMCache on a framework upgrade. This is the logic the real VM is doing in the node as well so shouldn't be something new.